ballerina-tests/tests/hateoas_types_test.bal

```diff
@@ -0,0 +1,182 @@
+        return {body: {message: "Links returned"}};
+    }
+
+    @http:ResourceConfig {
+        name: "link1"
+    }
+    resource function get link1() returns string {
+        return "Hello, from resource link1";
+    }
+
+    @http:ResourceConfig {
+        name: "link2"
+    }
+    resource function get link2() returns xml {
+        return xml`<message>Hello, from resource link2</message>`;
+    }
+
+    @http:ResourceConfig {
+        name: "link3"
+    }
+    resource function get link3() returns byte[] {
+        return "Hello, from resource link3".toBytes();
+    }
+
+    @http:ResourceConfig {
+        name: "link4"
+    }
+    resource function get link4() returns @http:Payload{mediaType: "application/json"} http:Response {
+        http:Response res = new;
+        res.setTextPayload("Hello, from resource link4");
+        return res;
+    }
+
+    @http:ResourceConfig {
+        name: "link5"
+    }
+    resource function get link5() returns http:Ok {
+        return {body: "Hello, from resource lin5"};
+    }
+
+    @http:ResourceConfig {
+        name: "link6"
+    }
+    resource function get lin6() returns string|xml {
+        return "Hello, from resource link6";
+    }
+
+    @http:ResourceConfig {
+        name: "link7"
+    }
+    resource function get link7() returns ReturnType {
+        return "Hello, from resource link7";
+    }
+
+    @http:ResourceConfig {
+        name: "link8"
+    }
+    resource function get link8() returns http:Response|error? {
+        return;
+    }
+
+    @http:ResourceConfig {
+        name: "link9"
+    }
+    resource function get link9() returns @http:Payload{mediaType: "application/json"} string|xml {
+        return xml`<message>Hello, from resource link9</message>`;
+    }
+
+    @http:ResourceConfig {
+        name: "link10"
+    }
+    resource function get link10() { }
+}
+
+
+final http:Client hateoasTypesTestClient = check new("http://localhost:" + hateoasTypesTestPort.toString());
+
+type HateoasTypesResponse record {|
+    *http:Links;
+    string message;
+|};
+
+@test:Config {}
+function testHateoasTypes() returns error? {
+    HateoasTypesResponse response = check hateoasTypesTestClient->get("/hateoas/links");
+    map<http:Link> expectedLinks = {
+        "link1": {
+            href: "/hateoas/link1",
+            types: [ "text/plain" ],
+            methods: [ http:GET ]
+        },
+        "link2": {
+            href: "/hateoas/link2",
+            types: [ "application/xml" ],
+            methods: [ http:GET ]
+        },
+        "link3": {
+            href: "/hateoas/link3",
+            types: [ "application/octet-stream" ],
+            methods: [ http:GET ]
+        },
+        "link4": {
+            href: "/hateoas/link4",
+            types: [ "application/json" ],
+            methods: [ http:GET ]
+        },
+        "link5": {
+            href: "/hateoas/link5",
+            types: [ "application/json" ],
```

Comment:
Prefer is we had skipped types in this link

Reply:
+1

Reply:
Fixed in [b808277](https://github.com/ballerina-platform/module-ballerina-http/pull/1270/commits/b808277a5738c9b3eee5ea594fe3b86fe9bd1a18)